docs/coroutine_test_examples.cpp

```diff
@@ -0,0 +1,598 @@
+
+    FunctionMemberObject fmo_test_move = std::move(fmo_test);
+    DataMemberObject dmo_test_move     = std::move(dmo_test);
+    NoMemberObject nmo_test_move       = std::move(nmo_test);
+
+    fmo_test_move.execute();
+    dmo_test_move.execute();
+    nmo_test_move.execute();
+
+    *data = -15;
+
+    fmo_test_move.execute();
+    dmo_test_move.execute();
+    nmo_test_move.execute();
+
+    // Output
+    // 99
+    // 99
+    // 99
+    // -14
+    // -14
+    // -14
+    // -15
+    // -15
+    // -15
+}
+
+void moveFromHeapToHeap()
+{
+    std::cout << std::endl << std::endl;
+    std::cout << "Test on the heap with moving variables" << std::endl;
+
+    auto data = std::make_shared<int>(99);
+
+    auto fmo_ptr = std::make_unique<FunctionMemberObject>(print_data_func, data);
+    auto dmo_ptr = std::make_unique<DataMemberObject>(print_data_func, data);
+    auto nmo_ptr = std::make_unique<NoMemberObject>(print_data_func, data);
+
+    fmo_ptr->execute();
+    dmo_ptr->execute();
+    nmo_ptr->execute();
+
+    *data = 3;
+
+    auto fmo_ptr_move = std::move(fmo_ptr);
+    auto dmo_ptr_move = std::move(dmo_ptr);
+    auto nmo_ptr_move = std::move(nmo_ptr);
+
+    fmo_ptr_move->execute();
+    dmo_ptr_move->execute();
+    nmo_ptr_move->execute();
+
+    *data = 4;
+
+    fmo_ptr_move->execute();
+    dmo_ptr_move->execute();
+    nmo_ptr_move->execute();
+
+    // Output
+    // 99
+    // 99
+    // 99
+    // 3
+    // 3
+    // 3
+    // 4
+    // 4
+    // 4
+}
+
+void moveFromStackToHeapFMO()
+{
+    std::cout << std::endl << std::endl;
+    std::cout << "Test moving from stack to heap - FunctionMemberObject" << std::endl;
+
+    auto data = std::make_shared<int>(99);
+
+    FunctionMemberObject fmo(print_data_func, data);
+
+    fmo.execute();
+
+    *data = 45;
+
+    auto fmo_heap = std::unique_ptr<FunctionMemberObject>();
+
+    *fmo_heap = std::move(fmo);
+
+    fmo_heap->execute();
+
+    *data = 83;
+
+    fmo_heap->execute();
+
+    // Output
+    // 99
+}
+
+void moveFromStackToHeapDMO()
+{
+    std::cout << std::endl << std::endl;
+    std::cout << "Test moving from stack to heap - DataMemberObject" << std::endl;
+
+    auto data = std::make_shared<int>(99);
+
+    DataMemberObject dmo(print_data_func, data);
+
+    dmo.execute();
+
+    *data = 45;
+
+    auto dmo_heap = std::unique_ptr<DataMemberObject>();
+
+    *dmo_heap = std::move(dmo);
+
+    dmo_heap->execute();
+
+    *data = 83;
+
+    dmo_heap->execute();
+
+    // Output
+    // 99
+}
+
+void moveFromStackToHeapNMO()
+{
+    std::cout << std::endl << std::endl;
+    std::cout << "Test moving from stack to heap - NoMemberObject" << std::endl;
+
+    auto data = std::make_shared<int>(99);
+
+    NoMemberObject nmo(print_data_func, data);
+
+    nmo.execute();
+
+    *data = 45;
+
+    auto nmo_heap = std::unique_ptr<NoMemberObject>();
+
+    *nmo_heap = std::move(nmo);
+
+    nmo_heap->execute();
+
+    *data = 83;
+
+    nmo_heap->execute();
+
+    // Output
```

Comment:
same as above

Reply:
printouts actually stop working at this point, so I've explicitly stated that